plugins/inputs/docker_log/docker_log.go

```diff
@@ -316,42 +313,23 @@ func (d *DockerLogs) tailContainerLogs(
 	}
 }
 
-func parseLogEntry(rawMessage []byte) (time.Time, []byte, error) {
-	var (
-		message   []byte
-		ts        []byte
-		err       error
-		timeStamp time.Time
-	)
+func parseLine(line []byte) (time.Time, string, error) {
+	parts := bytes.SplitN(line, []byte(" "), 2)
 
-	if len(rawMessage) == 0 {
-		return time.Time{}, nil, nil
+	switch len(parts) {
+	case 1:
+		parts = append(parts, []byte(""))
 	}
 
-	msgSplit := bytes.SplitN(rawMessage, []byte(" "), 2)
-
-	switch parts := len(msgSplit); {
-
-	case parts == 2: //regular case, timestamp separated from rawMessage with ' ': [timestamp] [rawMessage]
-		ts = msgSplit[0]
-		message = msgSplit[1]
-	case parts == 1:
-
-		if len(rawMessage) == dockerTimeStampLength { //edge case, empty message with only timestamp
-			ts = rawMessage
-			message = []byte{} //empty string
-		} else {
-			return time.Time{}, nil, fmt.Errorf("can't parse timestamp from log entry: %q", rawMessage)
-
-		}
-	}
+	tsString := string(parts[0])
+	message := string(parts[1])
 
-	timeStamp, err = time.Parse(time.RFC3339Nano, string(ts))
+	ts, err := time.Parse(time.RFC3339Nano, tsString)
 	if err != nil {
-		return time.Now(), message, fmt.Errorf("can't parse timestamp from string %q, reason: %v", ts, err)
+		return time.Time{}, "", fmt.Errorf("error parsing timestamp %q: %v", tsString, err)
 	}
 
-	return timeStamp, message, nil
+	return ts, message, nil
```

Comment:
@danielnelson , This cause copying the content of `message`, initially the function was returning `[]byte` that is cheaper from this perspective (especially respecting the possible long lines of output from multiple containers). Moreover, later, the message is trimmed with  `TrimRightFunc(message, unicode.IsSpace)`, that again involve building new string. Should we revert back this to using `[]bytes` until calling `acc.AddFields`? Do you agree?

Reply:
Let's move TrimRightFunc into this function and apply the bytes version of it before converting the message to a string.  The conversion/copy must happen at some point, its still best to do it here, but we just want to do it once.

Reply:
Adjusted as discussed